### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by jakyns**
- Add Dockerfile
- Add docker-compose.yml
- Add build.sh

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+RUN apt-get update \
+    && apt-get -y upgrade \
+    && apt-get install --yes postgresql-client \
+    && mkdir /app
+
+WORKDIR /app
+
+COPY . /app
+
+RUN mix local.hex --force
```
Comment (jakyns) [inline Dockerfile:12]: @teerawat1992 what is this doing wa?

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+mix deps.get
+
+mix ecto.create
+mix ecto.migrate
+
+mix test
```
Comment (jakyns) [inline build.sh]: @teerawat1992 shall we run test when building project?

File: config/dev.exs [hunk 3]
```diff
@@ -3,9 +3,9 @@ use Mix.Config
 # Configure your database
 config :ag, AG.Repo,
   username: "postgres",
-  password: "postgres",
+  password: "",
```
Comment (jakyns) [inline config/dev.exs]: @teerawat1992 please update these to read from env file hai gu noi

File: build.sh (hunk 2 above)
Comment (tteerawat) [reply build.sh]: No need

File: Dockerfile (hunk 1 above)
Comment (tteerawat) [reply Dockerfile:12]: install hex locally

File: config/dev.exs [hunk 4]
```diff
@@ -3,9 +3,9 @@ use Mix.Config
...
   database: "ag_dev",
-  hostname: "localhost",
+  hostname: "db",
```
Comment (tteerawat) [inline config/dev.exs]: Do we need to change this?

File: config/dev.exs (hunk 3 above)
Comment (tteerawat) [reply config/dev.exs]: How about simply set env variables in docker-compose instead

```
environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: ag_dev
```

Comment (teerawat1992) [reply config/dev.exs:6]: How about simply set env variables in docker-compose instead

```
environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: ag_dev
```

File: build.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+mix phx.server
```
Comment (tteerawat) [inline build.sh:8]: This is a simple build script which is fine :)
In the future we should consider using `mix release` instead 🙇

File: Dockerfile (hunk 1 above)
Comment (jakyns) [reply Dockerfile:12]: must or not wa?

File: config/dev.exs (hunk 4 above)
Comment (jakyns) [reply config/dev.exs]: docker-compose created its own network and it must be the same as this line https://github.com/jakyns/admiring-gamification/pull/8/files#diff-4e5e90c6228fd48698d074241c2ba760R7 that's why

File: config/dev.exs (hunk 3 above)
Comment (jakyns) [reply config/dev.exs]: it is fine but here should have changed to read from env vars, btw.

Comment (jakyns) [reply config/dev.exs]: just got it, deaw add hai.

**Comment by tteerawat** [comment]
AWESOME!

**Pull request merged**